# Introdução à Ciência de Dados - UFPB
Professor: Yuri Malheiros

## Medidas de Centralidade e Dispersão

A partir do dataset `movies.csv` calcule a média, mediana e desvio padrão para as colunas:

In [1]:
import os
import pandas as pd
from google.colab import drive

drive.mount('/gdrive')


Mounted at /gdrive


In [2]:
df = pd.read_csv('/gdrive/My Drive/Colab Notebooks/movies.csv', sep = ',')
df.head()

Title   US Gross  ...  IMDB Rating  IMDB Votes
0              The Land Girls   146083.0  ...          6.1      1071.0
1      First Love, Last Rites    10876.0  ...          6.9       207.0
2  I Married a Strange Person   203134.0  ...          6.8       865.0
3        Let's Talk About Sex   373615.0  ...          NaN         NaN
4                        Slam  1009819.0  ...          3.4       165.0

[5 rows x 16 columns]

### US Gross

In [13]:
# ADICIONE O SEU CÓDIGO AQUI
print("Média: ", df['US Gross'].mean())
print("Mediana: ", df['US Gross'].median())
print("Desvio padrão: ", df['US Gross'].std())
print("Diferença Média e mediana: ", abs(df['US Gross'].mean() - df['US Gross'].median()))



Média:  44002085.163744524
Mediana:  22019465.5
Desvio padrão:  62555311.39066204
Diferença Média e mediana:  21982619.663744524


### Production Budget

In [11]:
# ADICIONE O SEU CÓDIGO AQUI
print("Média: ", df['Production Budget'].mean())
print("Mediana: ", df['Production Budget'].median())
print("Desvio padrão: ", df['Production Budget'].std())
print("Diferença Média e mediana: ", abs(df['Production Budget'].mean() - df['Production Budget'].median()))

Média:  31069171.4484375
Mediana:  20000000.0
Desvio padrão:  35585913.44464446
Diferença Média e mediana:  11069171.4484375


### Running Time

In [12]:
# ADICIONE O SEU CÓDIGO AQUI
print("Média: ", df['Running Time (min)'].mean())
print("Mediana: ", df['Running Time (min)'].median())
print("Desvio padrão: ", df['Running Time (min)'].std())
print("Diferença Média e mediana: ", abs(df['Running Time (min)'].mean() - df['Running Time (min)'].median()))


Média:  110.19354838709677
Mediana:  107.0
Desvio padrão:  20.171013677535424
Diferença Média e mediana:  3.1935483870967687


### IMDB Rating

In [14]:
# ADICIONE O SEU CÓDIGO AQUI
print("Média: ", df['IMDB Rating'].mean())
print("Mediana: ", df['IMDB Rating'].median())
print("Desvio padrão: ", df['IMDB Rating'].std())
print("Diferença Média e mediana: ", abs(df['IMDB Rating'].mean() - df['IMDB Rating'].median()))




Média:  6.283467202141896
Mediana:  6.4
Desvio padrão:  1.2522899386004838
Diferença Média e mediana:  0.11653279785810433


### IMDB Votes

In [15]:
# ADICIONE O SEU CÓDIGO AQUI
print("Média: ", df['IMDB Votes'].mean())
print("Mediana: ", df['IMDB Votes'].median())
print("Desvio padrão: ", df['IMDB Votes'].std())
print("Diferença Média e mediana: ", abs(df['IMDB Votes'].mean() - df['IMDB Votes'].median()))





Média:  29908.64457831325
Mediana:  15106.0
Desvio padrão:  44937.58233504965
Diferença Média e mediana:  14802.644578313251


### Análise dos Resultados

Quais colunas tiveram valores muito diferentes entre média e mediana? Quais os seus desvios padrão?

* **US Gross**, std = 62555311.39066204
* **Production Budget**, std = 35585913.44464446
* **IMDB Votes**, std = 44937.58233504965

Que conclusões podemos tirar desses resultados?


As variáveis **US Gross, Production Budget e IMDB Votes** possuem a média distante da mediana, o que pode significar que existe muita distorção em seus dados (assimetria).

**Running Time e MDB Rating** Possuem a média proxima da mediana, o que pode significar que há pouca distorção em seus dados.




